docs-src/docs/guides/walkthrough.md

```diff
@@ -355,35 +399,432 @@ When you're finished with the example, you can instruct Terraform to destroy the
+	As previously stated, Terraform writes the latest status of your deployment to `terraform.tfstate`. As you move on to [Deployment 2](#2b-deploy-with-terraform) of this walkthrough, it will be easier for you to just destroy your Terraform deployment with `terraform destroy`.  
+    
+    However, the `apply` command is idempotent -- if you'd like, you may bring your `terraform.tfstate` file with you when you change directories to `iterative-walkthrough/deployment-2/`. When you run `terraform apply` in the next section of this walkthrough, Terraform will create new resources as expected - but it will also update existing resources that have changes and destroy resources that are no longer necessary, and update `terraform.tfstate` accordingly.
 
 
 ---
 
 ## Deployment 2: Guestbook + Redis
 
-INTRO TEXT GOES HERE. You can choose to complete this section using either the [Layer0 CLI](#2a-deploy-with-layer0-cli) or [Terraform](#2b-deploy-with-terraform).
+In this section, we're going to add some complexity to the previous deployment. [Deployment 1](#deployment-1-a-simple-guestbook-app) saw us create a simple guestbook application which kept its data in memory. But what if that ever came down, either by intention or accident? It would be easy enough to redeploy it, but all of the entered data would be lost. For this deployment, we're going to separate the data store from the guestbook application by creating a second Layer0 service which will house a local Redis database server and linking it to the first. You can choose to complete this section using either the [Layer0 CLI](#2a-deploy-with-layer0-cli) or [Terraform](#2b-deploy-with-terraform).
 
 
 ## 2a: Deploy with Layer0 CLI
 
+For this example, we'll be working in the `iterative-walkthrough/deployment-2/` directory.
+
+
+### Part 1: Create the Redis Load Balancer
+
+Both the Guestbook service and the Redis service will live in the same Layer0 environment, so we don't need to create one like we did in the first deployment. We'll start by making a load balancer behind which the Redis service will be deployed.
+
+The `Redis.Dockerrun.aws.json` task definition file we'll use is very simple - it just spins up a Redis server with the default configuration, which means that it will be serving on port 6379. Our load balancer needs to be able to forward TCP traffic to and from this port. And since the Redis server has no reason for anything other than the Guestbook application to be accessing it, we'll make this load balancer private. At the command prompt, execute the following:
```

Comment:
"since we don't want the redis server to be exposed to the public internet, we will put it behind a private load balancer. Private load balancers only accept traffic that originate from within their own environment`